maven_plugin/src/main/java/io/github/chains_project/maven_lockfile/AbstractLockfileMojo.java

```diff
@@ -77,32 +78,42 @@ protected Environment generateMetaInformation() {
     }
 
     protected AbstractChecksumCalculator getChecksumCalculator() throws MojoExecutionException {
-        ProjectBuildingRequest buildingRequest = new DefaultProjectBuildingRequest(session.getProjectBuildingRequest());
-        if (checksumMode.equals("maven_local")) {
-            return new FileSystemChecksumCalculator(dependencyResolver, buildingRequest, checksumAlgorithm);
-        } else if (checksumMode.equals("maven_central")) {
-            return new RemoteChecksumCalculator(checksumAlgorithm);
+        ProjectBuildingRequest artifactBuildingRequest = newResolveArtifactProjectBuildingRequest();
+        ProjectBuildingRequest pluginBuildingRequest = newResolvePluginProjectBuildingRequest();
+
+        checksumMode = checksumModeDeprication(checksumMode);
```

Comment:
```suggestion
        checksumMode = checksumModeDeprecation(checksumMode);
```
Make sure to fix this everywhere :)

Reply:
Thanks! Updated in [`ec3e90b` (#1151)](https://github.com/chains-project/maven-lockfile/pull/1151/commits/ec3e90b11fe6938199ebbeedde56e8f656b1b9ee).